### PR TITLE
Add a kubeadm ValidateNodeRegistrationOptions test

### PR DESCRIPTION
**Pull request by prasadkatti**
/sig cluster-lifecycle

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

/kind feature

**What this PR does / why we need it**:
Adds a missing test

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go [hunk 1]
```diff
@@ -111,6 +111,7 @@ func TestValidateNodeRegistrationOptions(t *testing.T) {
 		{"valid-nodename", kubeadmapiv1beta2.DefaultUrlScheme + "://" + "/some/path", false}, // supported, with socket url
 		{"valid-nodename", "bla:///some/path", true},                                         // unsupported url scheme
 		{"valid-nodename", ":::", true},                                                      // unparseable url
+		{"valid-nodename", "", false},                                                        // invalid CRISocket (path is not absolute)
```
Comment (neolit123) [inline cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go:114]: hi, if the path is not absolute an error should be returned:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/validation/validation.go#L535

why aren't we expecting an error (false)?

Comment (prasadkatti) [reply cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go]: hmm, weird. Maybe the changes in my workspace didn't get staged correctly. This should be true (expecting error). I must have changed it to false to simulate a failed test and that accidentally got staged.
Sorry about that. Updated and force-pushed.

**Comment by prasadkatti** [comment]
Closing this and making it a part of #91397

**Pull request closed without merging**